### PR TITLE
refactor(relay): use `RangeInclusive` to specify available ports

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

File: rust/relay/src/server.rs [hunk 1]
```diff
@@ -811,7 +810,7 @@ where
     }
 
     fn max_available_ports(&self) -> u16 {
-        self.highest_port - self.lowest_port
+        self.ports.end() - self.ports.start()
```
Comment (ReactorScram) [inline rust/relay/src/server.rs]: If it's inclusive and we had `[65535, 65535]` wouldn't that be 1 available port, not 0?

**Review by ReactorScram — APPROVED** [review]
Approving to unblock, just wondering about that `ports_available` count

File: rust/relay/src/server.rs (hunk 1 above)
Comment (ReactorScram) [reply rust/relay/src/server.rs]: It's a bit overkill but we could even use `ExactSizeIterator` for this https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html#method.len-3

Comment (thomaseizinger) [reply rust/relay/src/server.rs]: Well spotted! I ended up using the `.count` method on `Iterator`.

**Pull request merged**